portal/flatpak-portal.c

```diff
@@ -700,6 +741,479 @@ authorize_method_handler (GDBusInterfaceSkeleton *interface,
   return TRUE;
 }
 
+static void
+register_update_monitor (PortalFlatpakUpdateMonitor *monitor,
+                         const char *obj_path)
+{
+  G_LOCK (update_monitors);
+
+  g_hash_table_insert (update_monitors, g_strdup (obj_path), g_object_ref (monitor));
+
+  /* Trigger update timeout if needed */
+  if (update_monitors_timeout == 0 && !update_monitors_timeout_running_thread)
+    update_monitors_timeout = g_timeout_add_seconds (UPDATE_POLL_TIMEOUT_SEC, check_all_for_updates_cb, NULL);
```

Comment:
Why would that be strange? gmainloop is threadsafe, and it has to be, because its the main way to communicate between worker threads and the main thread in a typical UI app.

Reply:
Okay, I just didn't know the worker threads shared the same GMainLoop.